### PR TITLE
Improved OldDataMonitorTest.unlocatableRun

### PR DESCRIPTION
**Pull request by jglick**
# Description

Amends #1557 to avoid creating unversioned file `test/0/build.xml` discovered in #2851.

Tried backing out original fix to `referTo` and confirmed that the revised test fails.

### Changelog entries

None required.

### Desired reviewers

@reviewbybees & @recena

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -47,6 +47,7 @@ for(i = 0; i < buildTypes.size(); i++) {
                             def mvnCmd = "mvn -Pdebug -U clean install javadoc:javadoc ${runTests ? '-Dmaven.test.failure.ignore=true' : '-DskipTests'} -V -B -Dmaven.repo.local=${pwd()}/.repository" 
                             if(isUnix()) {
                                 sh mvnCmd
+                                sh 'test `git status --short | tee /dev/stderr | wc --bytes` -eq 0'
```
Comment (jglick) [inline Jenkinsfile]: inspired by something the NetBeans project did in CI IIRC

File: Jenkinsfile [hunk 2]
```diff
@@ -97,8 +98,6 @@ void withMavenEnv(List envVars = [], def body) {
     }
 }
 
-// This hacky method is used because File is not whitelisted,
-// so we can't use renameTo or friends
```
Comment (jglick) [inline Jenkinsfile]: You could not use them anyway since the workspace is on an agent. This comment is just misleading.

File: Jenkinsfile (hunk 1 above)
Comment (daniel-beck) [inline Jenkinsfile]: What is this supposed to do?

Comment (jglick) [inline Jenkinsfile]: See the second commit’s comment. Should have failed prior to the first commit.

File: .gitignore [hunk 3]
```diff
@@ -50,3 +50,6 @@ jenkins_*.changes
 push-build.sh
 war/node_modules/
 .java-version
+
+# cf. Jenkinsfile
+/.repository/
```
Comment (recena) [inline .gitignore]: @jglick Where it comes from?

File: Jenkinsfile (hunk 1 above)
Comment (recena) [inline Jenkinsfile]: @jglick What is it the relation with [this](https://github.com/jenkinsci/jenkins/pull/2851#discussion_r113501480)?

File: .gitignore (hunk 3 above)
Comment (daniel-beck) [inline .gitignore]: @recena Explained in the line above, specifically https://github.com/jenkinsci/jenkins/blob/master/Jenkinsfile#L44...L47

Comment (recena) [inline .gitignore]: @daniel-beck Thank Daniel. Difficult to see on a diff view.

Comment (jglick) [inline .gitignore]: Though actually I want to revert this and fix `Jenkinsfile` instead.

File: Jenkinsfile (hunk 1 above)
Comment (jglick) [inline Jenkinsfile]: That is explained by the first commit.

**Comment by jglick** [comment]
A Windows failure:

```
Expected: <0>
     but: was <1>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
	at hudson.util.ArgumentListBuilder2Test.echoArgs(ArgumentListBuilder2Test.java:120)
	at hudson.util.ArgumentListBuilder2Test.ensureArgumentsArePassedViaCmdExeUnmodified(ArgumentListBuilder2Test.java:88)
```

apparently due to

```
Error: Could not find or load main class hudson.util.EchoCommand
```

Looks offhand like a very fragile test sensitive to environment. Probably the `@` in the code source is confusing something in Windows.

**Comment by jglick** [comment]
Indeed I [made the test pass](https://ci.jenkins.io/job/Core/job/jenkins/job/PR-2860/5/testReport/hudson.util/ArgumentListBuilder2Test/ensureArgumentsArePassedViaCmdExeUnmodified_2/).

**Review by recampbell — APPROVED** [review]
LGTM :bee:

File: test/src/test/java/hudson/util/ArgumentListBuilder2Test.java [hunk 4]
```diff
@@ -92,12 +92,10 @@ public void ensureArgumentsArePassedViaCmdExeUnmodified() throws Exception {
     }
 
     public String echoArgs(String... arguments) throws Exception {
-        String testHarnessJar = Class.forName("hudson.util.EchoCommand")
+        String testHarnessJar = new File(Class.forName("hudson.util.EchoCommand")
```
Comment (jglick) [inline test/src/test/java/hudson/util/ArgumentListBuilder2Test.java]: Could also use `Paths.get(…)`, or `Which.jarFile`, but this seems to work well enough.

**Comment by jglick** [comment]
@reviewbybees done

**Review by olivergondza — APPROVED** [review]

**Review by stephenc — APPROVED** [review]

**Pull request merged**